binding/web/src/orca.ts

```diff
@@ -55,92 +68,311 @@ type pv_free_error_stack_type = (messageStack: number) => Promise<void>;
+        this._streamAddress = streamAddress;
+      }
+
+      /**
+       * Generates audio from text.
+       * The maximum number of characters per call to `.synthesize()` is `.maxCharacterLimit`.
+       * Allowed characters are lower-case and upper-case letters and punctuation marks that can be retrieved with `.validCharacters`.
+       * Custom pronunciations can be embedded in the text via the syntax `{word|pronunciation}`.
+       * The pronunciation is expressed in ARPAbet format, e.g.: "I {live|L IH V} in {Sevilla|S EH V IY Y AH}".
+       *
+       * @param text A string of text with properties described above.
+       */
+      public async synthesize(text: string): Promise<OrcaStreamSynthesizeResult> {
+        if (typeof text !== 'string') {
+          throw new OrcaErrors.OrcaInvalidArgumentError(
+            'The argument \'text\' must be provided as a string',
+          );
+        }
 
-    this._alignedAlloc = handleWasm.alignedAlloc;
-    this._wasmMemory = handleWasm.memory;
-    this._pvFree = handleWasm.pvFree;
-    this._objectAddress = handleWasm.objectAddress;
-    this._inputBufferAddress = handleWasm.inputBufferAddress;
-    this._messageStackAddressAddressAddress = handleWasm.messageStackAddressAddressAddress;
-    this._messageStackDepthAddress = handleWasm.messageStackDepthAddress;
+        if (text.trim().length > Orca._maxCharacterLimit) {
+          throw new OrcaErrors.OrcaInvalidArgumentError(
+            `'text' length must be smaller than ${Orca._maxCharacterLimit}`,
+          );
+        }
+
+        return new Promise<OrcaStreamSynthesizeResult>((resolve, reject) => {
+          this._Orca._functionMutex
+            .runExclusive(async () => {
+              if (this._Orca._wasmMemory === undefined) {
+                throw new OrcaErrors.OrcaInvalidStateError(
+                  'Attempted to call Orca stream synthesize after release.',
+                );
+              }
+
+              const memoryBufferText = new Uint8Array(this._Orca._wasmMemory.buffer);
+              const encodedText = new TextEncoder().encode(text);
+              const textAddress = await this._Orca._alignedAlloc(
+                Uint8Array.BYTES_PER_ELEMENT,
+                (encodedText.length + 1) * Uint8Array.BYTES_PER_ELEMENT,
+              );
+              if (textAddress === 0) {
+                throw new OrcaErrors.OrcaOutOfMemoryError(
+                  'malloc failed: Cannot allocate memory',
+                );
+              }
+              memoryBufferText.set(encodedText, textAddress);
+              memoryBufferText[textAddress + encodedText.length] = 0;
+
+              const numSamplesAddress = await this._Orca._alignedAlloc(
+                Int32Array.BYTES_PER_ELEMENT,
+                Int32Array.BYTES_PER_ELEMENT,
+              );
+              if (numSamplesAddress === 0) {
+                throw new OrcaErrors.OrcaOutOfMemoryError('malloc failed: Cannot allocate memory');
+              }
```

Comment:
can be created once used over and over again, clean up on delete

Reply:
should be multiple places we do this

Reply:
@albho theres multiples places it does this, I think you can just revert the change you did for this part since there's too many places and it doesn't really matter doing alloc here.